theme-kit/middleware.ts

```diff
@@ -0,0 +1,96 @@
+/* eslint-disable @typescript-eslint/no-use-before-define */
+import type { NextRequest } from 'next/server';
+import { NextResponse } from 'next/server';
+
+import { api } from './api';
+import { locale } from './locale';
+import { createRouter, route } from './router';
+
+export async function middleware(request: NextRequest) {
```

Comment:
The middleware, along with the router configuration, will become a part of the shared theme-kit logic that we won't need to reimplement in every theme.